src/pages/tasks/AssigneeSelectorModal.js

```diff
@@ -0,0 +1,192 @@
+
+    useEffect(() => {
+        const results = OptionsListUtils.getNewChatOptions(props.reports, props.personalDetails, props.betas, '', [], [], false);
+
+        setFilteredUserToInvite(results.userToInvite);
+        setFilteredRecentReports(results.recentReports);
+        setFilteredPersonalDetails(results.personalDetails);
+    }, [props]);
+
+    const updateOptions = useCallback(() => {
+        const {recentReports, personalDetails, userToInvite} = OptionsListUtils.getNewChatOptions(props.reports, props.personalDetails, props.betas, searchValue.trim(), [], [], false);
+
+        setHeaderMessage(OptionsListUtils.getHeaderMessage(recentReports?.length + personalDetails?.length !== 0, Boolean(userToInvite), searchValue));
+
+        setFilteredUserToInvite(userToInvite);
+        setFilteredRecentReports(recentReports);
+        setFilteredPersonalDetails(personalDetails);
+    }, [props, searchValue]);
+
+    useEffect(() => {
+        Timing.start(CONST.TIMING.SEARCH_RENDER);
+        Performance.markStart(CONST.TIMING.SEARCH_RENDER);
+
+        updateOptions();
+
+        return () => {
+            Timing.end(CONST.TIMING.SEARCH_RENDER);
+            Performance.markEnd(CONST.TIMING.SEARCH_RENDER);
+        };
+    }, [updateOptions]);
+
+    const debouncedUpdateOptions = _.debounce(updateOptions, 75);
+
+    const onChangeText = (newSearchTerm = '') => {
+        setSearchValue(newSearchTerm);
+        debouncedUpdateOptions();
+    };
+
+    const getSections = () => {
+        const sections = [];
+        let indexOffset = 0;
+
+        if (filteredRecentReports?.length > 0) {
+            sections.push({
+                title: props.translate('common.recents'),
+                data: filteredRecentReports,
+                shouldShow: true,
+                indexOffset,
+            });
+            indexOffset += filteredRecentReports?.length;
+        }
+
+        if (filteredPersonalDetails?.length > 0) {
+            sections.push({
+                data: filteredPersonalDetails,
+                shouldShow: true,
+                indexOffset,
+            });
+            indexOffset += filteredRecentReports?.length;
+        }
+
+        if (filteredUserToInvite) {
+            sections.push({
+                data: [filteredUserToInvite],
+                shouldShow: true,
+                indexOffset,
+            });
+        }
+
+        return sections;
+    };
+
+    const selectReport = (option) => {
+        if (!option) {
+            return;
+        }
+
+        if (option.reportID) {
+            setSearchValue('');
+
+            // eslint-disable-next-line no-console
+            console.log(option);
+
+            // TO-DO: Implement assigning the selected option to a state variable
+            // and navigate back to NewTaskPage
+
+            // Navigation.navigate(ROUTES.getReportRoute(option.text));
+        } else {
+            Report.navigateToAndOpenReport([option.login]);
+        }
+    };
```

Comment:
Ah yeah, setting it in state and then redirecting should be fine here - you seem to be on the right track


What search page looks like
```
    /**
     * Reset the search value and redirect to the selected report
     *
     * @param {Object} option
     */
    selectReport(option) {
        if (!option) {
            return;
        }

        if (option.reportID) {
            this.setState({
                searchValue: '',
            }, () => {
                Navigation.navigate(ROUTES.getReportRoute(option.reportID));
            });
        } else {
            Report.navigateToAndOpenReport([option.login]);
        }
    }
```